packages/devextreme/js/__internal/ui/selection/m_selection.ts

```diff
@@ -123,6 +124,10 @@ export default class Selection {
     this._selectionStrategy.setSelectedItems(keys, items);
   }
 
+  onSelectionChanging() {
```

Comment:
[hypothesis, excess code?]
I see that you never call this method in the Selection module. So possible it's excess and Selection should not have a `onSelectionChanging` at all? For the moment it seems to me like implementation on the Collection level is enough

Reply:
[b62f308](https://github.com/DevExpress/DevExtreme/pull/27860/commits/b62f30857d83924a9f3c3b9e14c53016e7d67bdb)